ajax/timeline.php

```diff
@@ -56,19 +56,19 @@
         $new_state = ($task->fields['state'] == Planning::DONE)
                         ? Planning::TODO
                         : Planning::DONE;
-        $new_label = Planning::getState($new_state);
-        echo json_encode([
-            'state'  => $new_state,
-            'label'  => $new_label
-        ]);
-
         $foreignKey = $parent->getForeignKeyField();
         $task->update([
             'id'        => intval($_POST['tasks_id']),
             $foreignKey => intval($_POST[$foreignKey]),
             'state'     => $new_state,
             'users_id_editor' => Session::getLoginUserID()
         ]);
+        $task->getFromDB(intval($_POST['tasks_id']));
```

Comment:
I'm not sure this is needed. Items automatically have `getFromDB` called after an update so the fields should already be up to date.

Reply:
Could you please provide the piece of code that can be used to reproduce this issue?

1. During call to `CommonDBTM::update()`, `$input` is copied into `$this->fields`, so `$this->fields` should reflect any input that has been "accepted" (i.e. not discarded by another plugin).
2. When update is done in DB, item is reloaded to get its new state from DB.

So, unless I miss something, if you do not have the expected `state` here, it means either it was discarded in `pre_update` process, either it was not saved in DB.

Reply:
Hi, 
In the plugin Behaviors _(setup.php)_, there is a call to beforeUpdate function from PluginBehaviorsTickettask class on pre-item-update event

> $PLUGIN_HOOKS['`pre_item_update`']['behaviors'] =
>       ['Problem'            => ['PluginBehaviorsProblem',           'beforeUpdate'],
>        'Ticket'             => ['PluginBehaviorsTicket',            'beforeUpdate'],
>        'ITILSolution'       => ['PluginBehaviorsITILSolution',      'beforeUpdate'],
>        `'TicketTask'         => ['PluginBehaviorsTickettask',        'beforeUpdate'],`
>        'ChangeTask'         => ['PluginBehaviorsChangetask',        'beforeUpdate'],
>        'ProblemTask'        => ['PluginBehaviorsProblemtask',       'beforeUpdate'],
>        'ProjectTask'        => ['PluginBehaviorsProjecttask',       'beforeUpdate'],
>        'PlanningExternalEvent' => ['PluginBehaviorsPlanningExternalEvent' , 'beforeUpdate']];

In _**/inc/tickettask.class.php**_, I added in **static function beforeUpdate(TicketTask $taskticket)** this part of code
```
        // récupération de la durée et du tech si passage par case à cocher		
        if (!isset($taskticket->input['actiontime'])) {
            $task = new TicketTask();
            $coche=1;
            if ($task->getFromDB($taskticket->input['id'])) {
               $taskticket->input['actiontime']=$task->fields['actiontime'];
               $taskticket->input['users_id_tech']=$task->fields['users_id_tech'];
               if ($task->fields['begin'] == 0) {
                  unset($coche);
               }
            }
        }
		
	// pas d'état fait si pas de durée ni date planifiée
       if ($taskticket->input['state'] == Planning::DONE && $taskticket->input['actiontime'] == 0 && !isset($taskticket->input['plan']['begin']) && !isset($taskticket->input['_planningrecall']) && !isset($coche)) {
            $taskticket->input['state'] = Planning::TODO;
            if (!isset($task)) {
                  Session::addMessageAfterRedirect(__("La durée est obligatoire sur une tâche faite",
                                                'behaviors'), true, ERROR);
            }
            return;
       }

      // pas d'état fait si pas de technicien 
      if ($taskticket->input['state'] == Planning::DONE && $taskticket->input['users_id_tech'] == 0) {
            $taskticket->input['state'] = Planning::TODO;
            if (!isset($task)) {
                  Session::addMessageAfterRedirect(__("Le technicien est obligatoire sur une tâche faite",
                                                'behaviors'), true, ERROR);
            }
            return;
         }
```